src/client/RunCodeWidget/index.tsx

```diff
@@ -40,14 +40,31 @@ export const RunCodeWidget = ({
 
     // Optional: reset the icon state after animation completes (e.g., 1 second)
     setTimeout(() => setIsRunning(false), 1000);
-  }, []);
+  }, [runCodeRef, runPrettierRef]);
 
   const handleSplitEditor = useCallback(() => {
     console.log('Split Editor');
     console.log(splitCurrentPane);
     splitCurrentPane();
   }, [splitCurrentPane]);
 
+  // Add the keyboard event listener for Ctrl+S and Shift+Enter
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if ((event.ctrlKey && event.key === 's') || (event.shiftKey && event.key === 'Enter')) {
```

Comment:
Very nice! I think there's some logic elsewhere that does this same check. Maybe a future refactor can unify these.